Formula/liquidctl.rb

```diff
@@ -0,0 +1,46 @@
+class Liquidctl < Formula
+  include Language::Python::Virtualenv
+
+  desc "Cross-platform tool and drivers for liquid coolers and other devices"
+  homepage "https://github.com/jonasmalacofilho/liquidctl"
+  url "https://files.pythonhosted.org/packages/source/l/liquidctl/liquidctl-1.3.0.tar.gz"
+  sha256 "ce0483b0a7f9cf2618cb30bdf3ff4195e20d9df6c615f69afe127f54956e42ce"
+
+  head "https://github.com/jonasmalacofilho/liquidctl.git"
+
+  depends_on "libusb"
+  depends_on "python"
+
+  resource "docopt" do
+    url "https://files.pythonhosted.org/packages/source/d/docopt/docopt-0.6.2.tar.gz"
+    sha256 "49b3a825280bd66b3aa83585ef59c4a8c82f2c8a522dbe754a8bc8d08c85c491"
+  end
+
+  resource "hidapi" do
+    url "https://files.pythonhosted.org/packages/source/h/hidapi/hidapi-0.7.99.post21.tar.gz"
+    sha256 "e0be1aa6566979266a8fc845ab0e18613f4918cf2c977fe67050f5dc7e2a9a97"
+  end
+
+  resource "pyusb" do
+    url "https://files.pythonhosted.org/packages/source/p/pyusb/pyusb-1.0.2.tar.gz"
+    sha256 "4e9b72cc4a4205ca64fbf1f3fff39a335512166c151ad103e55c8223ac147362"
+  end
+
+  def install
+    # customize liquidctl --version
+    ENV["DIST_NAME"] = "homebrew"
+    ENV["DIST_PACKAGE"] = "#{tap.nil? ? path : full_name} #{version}"
+
+    virtualenv_install_with_resources
+
+    man_page = buildpath/"liquidctl.8"
+    # setting the is_macos register to 1 adjusts the man page for macOS
+    inreplace man_page, ".nr is_macos 0", ".nr is_macos 1"
+    man.mkpath
+    man8.install man_page
+  end
+
+  test do
+    shell_output "#{bin}/liquidctl list --verbose --debug"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
This actually tests a _lot_ of stuff, as it runs the entire probe pipeline.  If a dependency is missing or if there's something majorly broken this will exit with non-zero status.

It's the best that can be done without assuming the availability of any specific hardware device.

---

Unit tests already run in CI for every upstream commit.

Reply:
To add a little more detail to this thread...

`liquidctl` monitors and controls fan speeds, voltages, and RGB lighting effects in a growing number of PC components such as all-in-one liquid coolers, fan controllers, smart power supplies, and lighting controllers. 

If the host computer does not contain a compatible device, then of course `liquidctl` is unable to function.

Therefore the test block in the Formula checks that the entire infrastructure is installed and ready should the user add a compatible device to their system.